Trimming fails if a conflict hits it

If conflict resolution doesn't resolve a conflict trimming will fail with an error like:
```
C:\Users\ericstj\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018: The "TrimFiles" task failed unexpectedly.\r [F:\testNetfx\sdkProj\sdkProj.csproj]
C:\Users\ericstj\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018: System.ArgumentException: An item with the same key has already been added. Key: System.Security.Cryptography.Encoding.dll\r [F:\testNetfx\sdkProj\sdkProj.csproj]
C:\Users\ericstj\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at System.ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException(Object key)\r [F:\testNetfx\sdkProj\sdkProj.csproj]
C:\Users\ericstj\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)\r [F:\testNetfx\sdkProj\sdkProj.csproj]
C:\Users\ericstj\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)\r [F:\testNetfx\sdkProj\sdkProj.csproj]
C:\Users\ericstj\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.TrimFiles.GetFiles(IDictionary`2 packages)\r [F:\testNetfx\sdkProj\sdkProj.csproj]
```

We shouldn't fail here, instead we should just tolerate the duplicate and use one of them.
https://github.com/dotnet/standard/blob/eacea174a2ea58c0dc8576cbec321e0cd44640a5/Microsoft.Packaging.Tools.Trimming/tasks/TrimFiles.cs#L310

I have encountered the same issue today:
```
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018: The "TrimFiles" task failed unexpectedly. [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018: System.ArgumentException: An item with the same key has already been added. Key: libSkiaSharp.dll [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at System.ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException(Object key) [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior) [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value) [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.TrimFiles.GetFiles(IDictionary`2 packages) [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.TrimFiles.Execute() [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
[00:02:54] C:\Users\appveyor\.nuget\packages\microsoft.packaging.tools.trimming\1.0.0-preview2-25401-01\build\Microsoft.Packaging.Tools.Trimming.targets(55,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() [C:\projects\core2d\apps\Core2D.Avalonia.NetCore\Core2D.Avalonia.NetCore.csproj]
```

--- a/Microsoft.Packaging.Tools.Trimming/tasks/TrimFiles.cs
+++ b/Microsoft.Packaging.Tools.Trimming/tasks/TrimFiles.cs
@@ -307,14 +307,16 @@ namespace Microsoft.DotNet.Build.Tasks
             foreach(var runtimeItem in runtimeItems)
             {
                 var fileNode = new FileNode(runtimeItem, packages);
-                files.Add(fileNode.Name, fileNode);
+
+                // last in wins
+                files[fileNode.Name] = fileNode;
             }
 
             // root files are likely not in the RuntimeItems
             foreach (var rootFile in RootFiles)
             {
                 var fileNode = new FileNode(rootFile, packages);
-                if (!files.ContainsKey(fileNode.Name)  && File.Exists(fileNode.SourceFile))
+                if (!files.ContainsKey(fileNode.Name) && File.Exists(fileNode.SourceFile))
                 {
                     files.Add(fileNode.Name, fileNode);
                 }

